GTMAppAuthSwift/Sources/KeychainConfiguration.swift

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+enum KeychainAttribute {
+  case dataProtectionKeychain
+  case accessGroup(String)
+
+  var keyName: String {
+    switch self {
+    case .dataProtectionKeychain:
+      return "kSecDataProtectionKeychain"
+    case .accessGroup:
+      return "kSecKeychainAccessGroup"
+    }
+  }
+}
+
+@objc(GTMKeychainConfiguration)
+public final class KeychainConfiguration: NSObject {
```

Comment:
If we're modeling Keychain attribute key/value pairs here, maybe this should be named `KeychainAttribute`?

Reply:
This class is actually not modeling the attributes. It's here because ObjC can't represent the `KeycainAttribute` enum. So, this class is here to wrap the attributes with an API that can configure `KeychainStore`. That's how I got to `KeychainConfiguration`.

Reply:
Updated `KeychainConfiguration` to be `KeychainAttribute` and moved `KeychainAttribute` enum to be nested type under the latter with name `Attribute`.